test/jdk/javax/swing/JMenuItem/8031573/bug8031573.java

```diff
@@ -20,45 +20,105 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+
+import java.awt.BorderLayout;
 import java.awt.FlowLayout;
-import javax.swing.JApplet;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.JButton;
 import javax.swing.JCheckBoxMenuItem;
+import javax.swing.JFrame;
 import javax.swing.JMenu;
 import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
 import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
+import javax.swing.text.JTextComponent;
 
 /* @test
  * @bug 8031573 8040279 8143064
  * @summary [macosx] Checkmarks of JCheckBoxMenuItems aren't rendered
  *           in high resolution on Retina
- * @author Alexander Scherbatiy
- * @run applet/manual=yesno bug8031573.html
+ * @run main/manual bug8031573
  */
-public class bug8031573 extends JApplet {
 
-    @Override
-    public void init() {
-        try {
+public class bug8031573 {
+
+    private static JFrame frame;
+    private static volatile boolean passed = false;
+    private static final CountDownLatch latch = new CountDownLatch(1);
 
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+    public static final String INSTRUCTIONS = "INSTRUCTIONS:\n\n"
+            + "Verify that high resolution system icons are used JCheckBoxMenuItem on HiDPI displays.\n"
+            + "If the display does not support HiDPI mode press PASS.\n"
+            + "1. Run the test on HiDPI Display.\n"
+            + "2. Press the Menu in the applet.\n"
+            + "3. Check that the icon on the JCheckBoxMenuItem is smooth If so, press PASS, else press FAIL.\n";
```

Comment:
```suggestion
            + "3. Check that the icon on the JCheckBoxMenuItem is smooth.\n"
            + "   If so, press PASS, otherwise press FAIL.";
```